src/chrome/logging/gdpr.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+/** Used to store data that can potentially be customer content. If it is, we should never log it or send it over telemetry */
+ export interface PossiblyCustomerContent<T> {
+    customerContentData: T;
+
+    /** Transform the data. If it is customer data, the result will sill be customer data */
+    transform<U>(transformFunction: (customerContentData: T) => U): PossiblyCustomerContent<U>;
+}
+
+/** Used to protect CustomerContent data so we don't inadvertently log it or send it as telemetry */
+export class CustomerContent<T> implements PossiblyCustomerContent<T> {
+    public constructor(data: T) {
+        Object.defineProperty(this, 'customerContentData', {
+            get: () => data
+        });
+    }
+
+    public static maybeNull<T>(data: T | null): PossiblyCustomerContent<T> | null {
+        return data !== null
+            ? new CustomerContent(data)
+            : null;
+    }
+
+    public static maybeUndefined<T>(data: T | undefined): PossiblyCustomerContent<T> | undefined {
+        return data !== undefined
+            ? new CustomerContent(data)
+            : undefined;
+    }
+
+    public static fromAsync<T>(promisedData: Promise<T>): Promise<PossiblyCustomerContent<T>> {
```

Comment:
This could also be made into a utility function

Reply:
I used to have 3 callers of this. I only have 1 now. Removing this utility function